compiler/main/checks.cpp

```diff
@@ -622,6 +624,16 @@ checkResolveRemovedPrims(void) {
   }
 }
 
+static void checkNoRecordDeletes() {
+  // No need to do for_alive_in_Vec - there shouldn't be any, period.
+  // User errors are to be detected by chpl__delete() in the modules.
+  forv_Vec(CallExpr, call, gCallExprs)
+    if (FnSymbol* fn = call->resolvedFunction())
+      if(fn->hasFlag(FLAG_DESTRUCTOR))
+        if (isRecord(call->get(1)->typeInfo()->getValType()))
+          INT_FATAL(call, "delete on records");
```

Comment:
Ah good question. Generally, INT_FATAL and INT_ASSERT are triggered by bugs in the compiler. "INT" stands for "internal [to the compiler]". Cf. USR_ stands for "user".

For this one, I am not aware of a bug that would trigger this INT_FATAL. So I do not know how to test it. The same is true for the great majority of INT_FATAL and INT_ASSERT calls in the compiler.

It's like an assert() in C code - it's there to ensure that it is never triggered.